File: Computational_Physics_2/1.Lectures/8.Computo_en_paralelo_OPENMP/examples/example_1_Hello_World/C/omp_hello.c
/*  C  */

#include <stdio.h>
#include <unistd.h>
#include <omp.h>

int main () {

    /* Serial Region  (master thread)                                */
    /* Parameters of the Application                                 */
    int len=30;
    char name[30];

    /* OpenMP Parameters */
    int id, nthreads;

    /* Master thread obtains information about itself and its environment. */
    nthreads = omp_get_num_threads();       /* get number of threads */
    id = omp_get_thread_num();              /* get thread            */
    gethostname(name,len);                  /* get run-host name     */
    printf("SERIAL REGION:     Runhost:%s   Thread:%d of %d thread    hello, world\n", name,id,nthreads);

    /* Open parallel region. */
    /* Each thread obtains information about itself and its environment. */
    #pragma omp parallel private(name,id,nthreads)
    {nthreads = omp_get_num_threads();      /* get number of threads */
     id = omp_get_thread_num();             /* get thread            */
     gethostname(name,len);                 /* get run-host name     */
     printf("PARALLEL REGION:   Runhost:%s   Thread:%d of %d threads   hello, world\n", name,id,nthreads);
    }
    /* Close parallel region. */

    /* Serial Region  (master thread)                                */
    printf("SERIAL REGION:     Runhost:%s   Thread:%d of %d thread    hello, world\n", name,id,nthreads);
    return 0;
}

File: Computational_Physics_2/1.Lectures/8.Computo_en_paralelo_OPENMP/examples/example_1_Hello_World/Fortran/omp_hello.f90
!  Fortran 90

PROGRAM hello
    use omp_lib          ! Fortran 90; omp_get_thread_num, omp_get_num_threads

    ! Serial Region  (master thread)
    ! Parameters of the Application
    CHARACTER(30) name   ! Fortran 90

    ! OpenMP Parameters
    INTEGER id, nthreads

    ! Master thread obtains information about itself and its environment.
    nthreads = omp_get_num_threads()       ! get number of threads
    id = omp_get_thread_num()              ! get thread
    CALL getenv("HOST",name)               ! get run-host name
    WRITE (*,*) 'SERIAL REGION:   Runhost:', name, '   Thread:', id, ' of ', nthreads, ' thread    hello, world'

    ! Open parallel region.
    ! Each thread obtains information about itself and its environment.
    !$OMP PARALLEL PRIVATE(name,id,nthreads)
        nthreads = omp_get_num_threads()   ! get number of threads
        id = omp_get_thread_num()          ! get thread
        CALL getenv("HOST",name)           ! get run-host name
        WRITE (*,*) 'PARALLEL REGION:   Runhost:', name, '   Thread:', id, ' of ', nthreads, ' threads   hello, world'
    !$OMP END PARALLEL
    ! Close parallel region.

    ! Serial Region  (master thread)
    WRITE (*,*) 'SERIAL REGION:   Runhost:', name, '   Thread:', id, ' of ', nthreads, ' thread    hello, world'
END PROGRAM hello        ! Fortran 90